File: Dailo/Dailo.ClientApp/src/app/modules/habits/models/habit.model.ts
import { FrequencyModel } from '@habits/models/frequency.model';
import { TargetModel } from '@habits/models/target.model';
import { HabitType } from '@habits/enums/habit-type.enum';
import { HabitStatus } from '@habits/enums/habit-status.enum';
import { AutomationSource } from '@habits/enums/automation-source.enum';
import { MilestoneModel } from '@habits/models/milestone.model';
import { TagModel } from '../../tags/models/tag.model';

export interface HabitModel {
  id: string;
  name: string;
  description?: string;
  type: HabitType;
  frequency: FrequencyModel;
  target: TargetModel;
  status: HabitStatus;
  isArchived: boolean;
  endDate?: Date;
  milestone?: MilestoneModel;
  automationSource?: AutomationSource;
  tags: TagModel[];

  createdAtUtc: Date;
  lastCompletedAtUtc: Date;
}

File: Dailo/Habit.Application/Models/HabitModel.cs
using Habit.Domain.Enums;
using StrictId;

namespace Habit.Application.Models;

public class HabitModel
{
    public required Id<HabitModel> Id { get; set; }

    public required string Name { get; set; }

    public string? Description { get; set; }

    public required HabitType Type { get; set; }

    public required FrequencyModel Frequency { get; set; }

    public required TargetModel Target { get; set; }

    public required HabitStatus Status { get; set; }

    public bool IsArchived { get; set; }

    public DateOnly? EndDate { get; set; }

    public MilestoneModel? Milestone { get; set; }

    public DateTime CreatedAtUtc { get; set; }

    public DateTime? LastCompletedAtUtc { get; set; }

    public IEnumerable<TagModel> Tags { get; set; } = [];
}
